National Library of Norway.js

```diff
@@ -124,38 +199,82 @@ function scrape(doc, url) {
 				;
 		});
 
-		if (item.numPages) {
-			item.numPages = extractNumPages(item.numPages);
-		}
-		
 		item.date = ZU.strToISO(item.date);
 
-		// Add permalink
-		var container = doc.getElementById('preview_metadata');
-		if (container) {
-			item.url = ZU.xpathText(container, './/a[contains(@href, "urn.nb.no")]');
-		}
+		// Use the (undocumented) JSON api to add some data missing in the RIS export
+		apiRequest(apiUrl, function(apiResponse) {
+
+			var m = apiResponse.metadata;
+
+			item.title = m.title;
+			item.shortTitle = m.titleInfos.title;
+			item = trimBrackets(item);
+
+			item.numPages = extractNumPages(m.physicalDescription.extent);
+
+			if (m.identifiers.isbn13 && m.identifiers.isbn13.length) {
+				item.ISBN = m.identifiers.isbn13[0];
+			} else if (m.identifiers.isbn10 && m.identifiers.isbn10.length) {
+				item.ISBN = m.identifiers.isbn10[0];
+			}
+
+			if (item.accessDate) {
+				// Better leave this to Zotero
+				delete item.accessDate;
+			}
+
+			if (m.identifiers.urn) {
+				item.url = 'https://urn.nb.no/' + apiResponse.metadata.identifiers.urn;
```

Comment:
just making sure: all items with a URN are actual digital objects (like e-books) at that URL, right? If so, this is great. If not, we'd want to attach them as links, though.

Reply:
That's my understanding, in the form of digitized material. Most Norwegian books have been scanned, but those not out of copyright are IP restricted to either Norway or the venues of the National Library, so you might end up with just a metadata record in practice, but I think the principle is that the URN lead to the digitized version.